### PR TITLE
Add special characters missing test coverage for native-fs s3 path

### PR DESCRIPTION
**Pull request by anusudarsan**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by martint** [comment]
Can't we do this as a regular JUnit test? Why does it need to be a product test? 

cc @electrum

**Comment by electrum** [comment]
We should convert the existing tests to use native. We don’t need more product tests. The only usefulness of a product test is to verify that the feature works at all (such as due to some class loader issue). It doesn’t need to test full functionality — that should be done in unit tests.

**Comment by anusudarsan** [comment]
updated

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestHiveS3MinioQueries.java [hunk 1]
```diff
@@ -26,29 +26,37 @@
...
-                        .put("hive.s3.path-style-access", "true")
+                        .put("fs.hadoop.enabled", "false")
+                        .put("fs.native-s3.enabled", "true")
+                        .put("hive.metastore", "thrift")
+                        .put("hive.metastore.uri", hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint().toString())
+                        .put("s3.aws-access-key", MINIO_ACCESS_KEY)
+                        .put("s3.aws-secret-key", MINIO_SECRET_KEY)
+                        .put("s3.region", MINIO_REGION)
+                        .put("s3.endpoint", hiveMinioDataLake.getMinio().getMinioAddress())
+                        .put("s3.path-style-access", "true")
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestHiveS3MinioQueries.java:59]: We could remove these properties by using `S3HiveQueryRunner.builder(hiveMinioDataLake)`.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestHiveS3MinioQueries.java [hunk 2]
```diff
@@ -105,4 +113,44 @@ private String getDeclaredTableLocation(String tableName)
...
+                        "(3, 'with:colon')," +
+                        "(4, 'with/slash')," +
+                        "(5, 'with\\\\backslashes')," +
+                        "(6, 'with\\backslash')," +
+                        "(7, 'with=equal')," +
+                        "(8, 'with?question')," +
+                        "(9, 'with!exclamation')," +
+                        "(10, 'with%percent')," +
+                        "(11, 'with%%percents')," +
+                        "(12, 'with space')");
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestHiveS3MinioQueries.java:153]: We could extract a variable for the same values in INSERT and SELECT statements.

**Review by ebyhr — APPROVED** [review]

**Pull request merged**